### PR TITLE
fix: do not use pointers for env maps

### PR DESCRIPTION
**Pull request by ctrox**
using pointers breaks kong's arg parsing somewhat as it fails to merge multiple values of the same arg. Removing the pointer fixes that while still making it possible to detect if the arg has been supplied or not as a map has nil as the zero value.

### CODE REVIEW

File: update/application.go [hunk 1]
```diff
@@ -180,7 +180,7 @@ func (cmd *applicationCmd) applyUpdates(app *apps.Application) {
 
 	var env map[string]string
 	if cmd.Env != nil {
-		env = *cmd.Env
+		env = cmd.Env
```
Comment (thirdeyenick) [inline update/application.go:183]: Can't we then get rid of the `env` variable and use `cmd.Env` directly? The `env` variable will also be `nil` if `cmd.Env` is `nil`.

Comment (thirdeyenick) [reply update/application.go:183]: But we do this already. If `cmd.Env` is `nil` (which it is if it is not set, right?), then the `env` var is also `nil` (as we don't use `make`).

Comment (ctrox) [reply update/application.go]: Yeah you are right, we don't make the make there, we just do it for the `buildEnv`.

**Review by thirdeyenick — APPROVED** [review]
👍

**Pull request merged**